src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -65,9 +66,15 @@ class PremisesController(
   private val staffMemberTransformer: StaffMemberTransformer,
   private val staffMemberService: StaffMemberService
 ) : PremisesApiDelegate {
-  override fun premisesGet(): ResponseEntity<List<Premises>> {
+  override fun premisesGet(service: ServiceCategory?): ResponseEntity<List<Premises>> {
     return ResponseEntity.ok(
-      premisesService.getAllPremises().map {
+      premisesService.getAllPremises().filter {
```

Comment:
Maybe premature optimisation but this would be better handled by the SQL server I think - in the PremisesRepository interface (lives in `src/main/kotlin/..../jpa/entity/PremisesEntity.kt` if you add a method like this:

```
@Repository
interface PremisesRepository : JpaRepository<PremisesEntity, UUID> {
  fun findAllByType(type: String): List<PremisesEntity>
}
```

spring-data-jpa will do the query for you (see https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details)

Reply:
This has been implemented.